bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+
+          cat /signer-ca/tls.crt | openssl x509 -outform pem -text > /etc/openvswitch/keys/ipsec-cacert.pem
+
+          ovs-vsctl --retry -t 60 set Open_vSwitch . other_config:certificate=/etc/openvswitch/keys/ipsec-cert.pem \
+                                                     other_config:private_key=/etc/openvswitch/keys/ipsec-privkey.pem \
+                                                     other_config:ca_cert=/etc/openvswitch/keys/ipsec-cacert.pem
+
+          popd
+
+          # Workaround for https://github.com/libreswan/libreswan/issues/373
+          ulimit -n 1024
+
+          /usr/libexec/ipsec/addconn --config /etc/ipsec.conf --checkconfig
+          # Check for kernel modules
+          /usr/libexec/ipsec/_stackmanager start
+          # Check for nss database status and migration
+          /usr/sbin/ipsec --checknss
+          # Check for nflog setup
+          /usr/sbin/ipsec --checknflog
+          # Start the actual IKE daemon
+          /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf --logfile /var/log/openvswitch/libreswan.log
+
+          # Workaround for https://mail.openvswitch.org/pipermail/ovs-dev/2020-October/375734.html
+          OVS_LOGDIR=/var/log/openvswitch OVS_RUNDIR=/var/run/openvswitch OVS_PKGDATADIR=/usr/share/openvswitch /usr/share/openvswitch/scripts/ovs-ctl --ike-daemon=libreswan start-ovs-ipsec
+
+          # Wait for the nbdb container in the ovnkube-master pod to become ready before we try to write to
+          # it.
+          if kubectl wait --for=condition=Ready pod -l app=ovnkube-master -n ${ovn_kubernetes_namespace} --timeout 600s; then
+            echo "ovnkube-master is not Ready, unable to write to OVNNB DB"
+            exit 2
+          else
+            ovn-nbctl set nb_global . ipsec=true
+          fi
+
+          while true; do
+            sleep 10
+          done
+        env:
+        - name: OVS_LOG_LEVEL
+          value: info
+        - name: K8S_NODE
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - mountPath: /var/run/openvswitch
+          name: host-var-run-ovs
+        - mountPath: /var/log/openvswitch/
+          name: host-var-log-ovs
+        - mountPath: /var/run/ovn/
+          name: host-var-run-ovn
+        - mountPath: /signer-ca
+          name: signer-ca
+        - mountPath: /etc/openvswitch
+          name: etc-openvswitch
+        resources:
+        # TODO what is reasonable here?
+          requests:
+            cpu: 100m
+            memory: 300Mi
+          limits:
+            cpu: 200m
+            memory: 400Mi
+        terminationMessagePolicy: FallbackToLogsOnError
+        readinessProbe:
+          exec:
+            command:
+            - /bin/bash
+            - -c
+            - |
+              #!/bin/bash
+              ovs-appctl -t ovs-monitor-ipsec ipsec/status
```

Comment:
Do we need to parse the output and validate a specific status? Or is it fine if this just returns the exit code 0?

Reply:
I did do that initially but it was causing me problems due to another issue. I will change it back.

Reply:
On second thoughts, I think for the moment this is sufficient as this will simultaneously test if ovs-monitor-ipsec and libreswan are running.